### PR TITLE
OCPBUGS-31398: Recycler-pod image now points to the OCP Payload reference

### PR DESCRIPTION
**Pull request by jparrill**
**Which issue(s) this PR fixes**:
Fixes #[OCPBUGS-31398](https://issues.redhat.com/browse/OCPBUGS-31398)

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/kcm/config.go [hunk 1]
```diff
@@ -58,13 +62,20 @@ func generateConfig(serviceServingCA *corev1.ConfigMap) (string, error) {
...
+func ReconcileRecyclerConfig(config *corev1.ConfigMap, ownerRef config.OwnerRef, releaseImageProvider *imageprovider.ReleaseImageProvider) error {
+	var result strings.Builder
+
 	ownerRef.ApplyTo(config)
 	if config.Data == nil {
 		config.Data = map[string]string{}
 	}
+
+	data := map[string]string{
+		"rhtoolsImageName": releaseImageProvider.GetImage("tools"),
```
Comment (Patryk-Stefanski) [inline control-plane-operator/controllers/hostedcontrolplane/kcm/config.go:74]: should we add a check to make sure that the image exists?

Comment (jparrill) [reply control-plane-operator/controllers/hostedcontrolplane/kcm/config.go:74]: I don't think that's necessary meanwhile it's part of the base in the ocp payload.

**Pull request merged**